mixed-content/generic/common.js

```diff
@@ -0,0 +1,238 @@
+function normalizePort(targetPort) {
+  var defaultPorts = [80, 443];
+  var isDefaultPortForProtocol = (defaultPorts.indexOf(targetPort) >= 0);
```

Comment:
This tells you if a port is a default port, but not if it's the right default port. That is, `http://example.com:443` would normalize to "" with this code, which is incorrect.